terraform/files/template/cluster-template.yaml

```diff
@@ -56,7 +56,7 @@ spec:
         kubeletExtraArgs:
           cloud-provider: external
     clusterConfiguration:
-      imageRepository: k8s.gcr.io
+      imageRepository: registry.k8s.io
```

Comment:
```suggestion
      imageRepository: registry.k8s.io
      dns:
        imageRepository: registry.k8s.io/coredns
```

Reply:
I added these changes, but that doesn't change the fact that I can't test it. For some reason the OS_CLOUD environment variable just isn't set correctly for later scripts.
For the sake of time here, I think it would be better if you could test the affected versions, then this would probably resolved faster.

Reply:
As I wrote, for me all the versions work. Can you try running `make fullclean` and try it from the beginning?

Reply:
`make fullclean` helped with the wrong `OS_CLOUD` env variable, but the `testcluster` still isn't created.
It just waits endlessly for the cluster to  be `READY` and then outputs:
`openstack_compute_instance_v2.mgmtcluster_server (remote-exec): error: timed out waiting for the condition on kubeadmcontrolplanes/testcluster-control-plane`

Reply:
is your mgmtcluster instance running? can you ssh into it? `make login USERNAME=ubuntu` and check capi/capo components logs why it can't create a control-plane node(I understand that you do not see control-plane OpenStack instance - there is only one mgmtcluster instance)

Reply:
The `ubuntu-capi-image-${v}` is created correctly. I can also ssh onto the mgmtcluster instance. There, the logs of the capi/capo components produced the following output:

capi-controller-manager
```
I0622 08:42:44.338458       1 machine_controller_phases.go:219] "Waiting for bootstrap provider to generate data secret and report status.ready" controller="machine" controllerGroup="cluster.x-k8s.io" controllerKind="Machine" Machine="default/capi-testcluster-md-0-no1-58fc4dccc6-57dq6" namespace="default" name="capi-testcluster-md-0-no1-58fc4dccc6-57dq6" reconcileID=71b5d18d-d14c-42a3-bab5-0acf3849846c MachineSet="default/capi-testcluster-md-0-no1-58fc4dccc6" MachineDeployment="default/capi-testcluster-md-0-no1" Cluster="default/testcluster" KubeadmConfig="default/capi-testcluster-md-0-genw01-5pphm"
I0622 08:45:06.305204       1 machine_controller_phases.go:286] "Waiting for infrastructure provider to create machine infrastructure and report status.ready" controller="machine" controllerGroup="cluster.x-k8s.io" controllerKind="Machine" Machine="default/capi-testcluster-md-0-no1-58fc4dccc6-fpb8c" namespace="default" name="capi-testcluster-md-0-no1-58fc4dccc6-fpb8c" reconcileID=85bbc376-e9f5-4b70-a987-fd0ed9dbf106 MachineSet="default/capi-testcluster-md-0-no1-58fc4dccc6" MachineDeployment="default/capi-testcluster-md-0-no1" Cluster="default/testcluster" OpenStackMachine="default/capi-testcluster-md-0-genw01-qctm2"
I0622 08:45:06.305429       1 machine_controller_noderef.go:49] "Waiting for infrastructure provider to report spec.providerID" controller="machine" controllerGroup="cluster.x-k8s.io" controllerKind="Machine" Machine="default/capi-testcluster-md-0-no1-58fc4dccc6-fpb8c" namespace="default" name="capi-testcluster-md-0-no1-58fc4dccc6-fpb8c" reconcileID=85bbc376-e9f5-4b70-a987-fd0ed9dbf106 MachineSet="default/capi-testcluster-md-0-no1-58fc4dccc6" MachineDeployment="default/capi-testcluster-md-0-no1" Cluster="default/testcluster" OpenStackMachine="default/capi-testcluster-md-0-genw01-qctm2"
```

capo-controller-manager
```
**E0622 08:44:25.362487       1 controller.go:326] "Reconciler error" err="failed to create load balancer service client: No suitable endpoint could be found in the service catalog." controller="openstackcluster" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackCluster" OpenStackCluster="default/testcluster" namespace="default" name="testcluster" reconcileID=7c877dcc-7ce6-4e38-8763-1ea61344044e**
I0622 08:47:42.374596       1 openstackmachine_controller.go:311] "Cluster infrastructure is not ready yet, requeuing machine" controller="openstackmachine" controllerGroup="infrastructure.cluster.x-k8s.io" controllerKind="OpenStackMachine" OpenStackMachine="default/capi-testcluster-md-0-genw01-sz8gs" namespace="default" name="capi-testcluster-md-0-genw01-sz8gs" reconcileID=805f6db7-9e1b-43dd-b174-0beba1801fbf openStackMachine="capi-testcluster-md-0-genw01-sz8gs" machine="capi-testcluster-md-0-no1-58fc4dccc6-57dq6" cluster="testcluster" openStackCluster="testcluster"
```

capi-kubeadm-control-plane
```
E0622 08:52:01.367937       1 controller.go:185] "Failed to update KubeadmControlPlane Status" err="failed to create remote cluster client: failed to retrieve kubeconfig secret for Cluster default/testcluster: secrets \"testcluster-kubeconfig\" not found" controller="kubeadmcontrolplane" controllerGroup="controlplane.cluster.x-k8s.io" controllerKind="KubeadmControlPlane" KubeadmControlPlane="default/testcluster-control-plane" namespace="default" name="testcluster-control-plane" reconcileID=f3713b0c-a551-4eee-a261-4f083ff621db Cluster="default/testcluster"
```
I marked the interesting part, which is probably the problem with the load-balancer endpoint.
**After talking with our ITO team, the OpenStack I currently have my project in doesn't have the endpoint (no Octavia is provided).**

Reply:
Yeah, it is probably a problem with your OpenStack. 

If you want to test it, I suggest you create a cluster without a load balancer, you can do it by commenting out these [lines](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/blob/main/terraform/files/template/cluster-template.yaml#L33-L34) before `make create`.